### ISSUE TITLE
VSCode fails to start RLS: '--toolchain <toolchain>' requires a value but none was supplied

### ISSUE THREAD

**Issue by tYYGH** [rust-lang/rls-vscode#689]
I am learning rust, and only beginning. My setup:

OS: up-to-date 64-bit Archlinux with `rustup 1.20.2` and `code 1.39.2` installed from the software repositories.
Commands run so far:

```bash
$ rustup default stable
$ rustup component add rls rust-analysis rust-src
```
In VS Code :

```
Command failed: rustup component list --toolchain error: The argument '--toolchain <toolchain>' requires a value but none was supplied USAGE: rustup component list --toolchain <toolchain> For more information try --help
```

**Comment by tYYGH** [comment]
It seems, that VS Code is not able to deal with custom paths.
In my environment, I have:

```
CARGO_HOME="$HOME/.local/opt/rust_cargo"
RUSTUP_HOME="$HOME/.local/opt/rustup"
PATH="$HOME/.local/bin:$CARGO_HOME/bin:…"
```

And VS Code is obviously ignoring both `…_HOME` variables. I _can_ set `CARGO_HOME` in the settings, but even then, it is ignored. `RUSTUP_HOME` cannot even be set. Anyway, VS Code keeps creating `~/.cargo` and `~/.rustup`… I tried symlinking these to my custom places so that VS Code finds them, but even so, it says that RLS is not found.

**Comment by TeHMoroS** [comment]
I've had the same problem, but after some fiddling around, ignoring the issue and proceeding in configuring my Webpack setup, the plugin involved in Webpack handling of Rust-related code (wasm-pack-plugin) started complaining about the lack of a default toolchain.

TL;DR After executing the following commands to download and set the default stable toolchain:
```
rustup install stable
rustup default stable
```
all problems went away and VSCode is not complaining anymore.

**Comment by Dieterbe** [comment]
@TeHMoroS thanks for that. i was running into the error from OP on arch linux, running those 2 commands fixed it.

**Comment by ramonGonzEdu** [comment]
Thanks @TeHMoroS That worked perfectly

**Comment by ramonGonzEdu** [comment]
nevermind

**Comment by infiniteregrets** [comment]
Had the same issue on MacOS, but reinstalling rustup fixes it. One of the comments on https://github.com/rust-lang/vscode-rust/issues/622 mentions adding `"rust-client.rustupPath": "$HOME/.cargo/bin/rustup"` to the settings.json file but that didn't work for me
